src/aind_metadata_mapper/fib/session.py

```diff
@@ -0,0 +1,295 @@
+"""Module for creating Fiber Photometry session metadata.
+
+This module implements an ETL (Extract, Transform, Load) pattern for generating
+standardized session metadata from fiber photometry experiments. It handles:
+
+- Extraction of session times from data files
+- Transformation of raw data into standardized session objects
+- Loading/saving of session metadata in a standard format
+
+The ETL class provides hooks for future extension to fetch additional data from
+external services or handle new data formats.
+"""
+
+import sys
+import json
+from typing import Union
+import logging
+
+from aind_data_schema.core.session import (
+    DetectorConfig,
+    FiberConnectionConfig,
+    LightEmittingDiodeConfig,
+    Session,
+    Stream,
+)
+from aind_data_schema_models.modalities import Modality
+
+from aind_metadata_mapper.core import GenericEtl
+from aind_metadata_mapper.core_models import JobResponse
+from aind_metadata_mapper.fib.models import JobSettings
+from aind_metadata_mapper.fib.utils import (
+    extract_session_start_time_from_files,
+    extract_session_end_time_from_files,
+    verify_output_file,
+)
+
+
+class ETL(GenericEtl[JobSettings]):
+    """Creates fiber photometry session metadata using an ETL pattern.
+
+    This class handles the full lifecycle of session metadata creation:
+    - Extracting timing information from data files
+    - Transforming raw data into standardized session objects
+    - Loading/saving session metadata in a standard format
+
+    The ETL process ensures that all required metadata fields are populated
+    and validates the output against the AIND data schema.
+    """
+
+    def __init__(self, job_settings: Union[str, JobSettings]):
+        """Initialize ETL with job settings.
+
+        Args:
+            job_settings: Either a JobSettings object or a JSON string that can
+                be parsed into one. The settings define all required parameters
+                for the session metadata, including experimenter info, subject
+                ID, data paths, etc.
+
+        Raises:
+            ValidationError: If the provided settings fail schema validation
+            JSONDecodeError: If job_settings is a string but not valid JSON
+        """
+        if isinstance(job_settings, str):
+            job_settings = JobSettings(**json.loads(job_settings))
+        super().__init__(job_settings)
+
+    def _extract(self) -> JobSettings:
```

Comment:
The extract methods usually parse the input sources into intermediate data models that can be transformed into the output Session object.

Reply:
I made some heavy modifications to session.py. Let me know if that satisfies you.